server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -349,18 +347,24 @@ public Query rangeQuery(
             }
 
             @Override
-            public List<Field> createFields(String name, Number value, boolean indexed, boolean docValued, boolean stored) {
-                List<Field> fields = new ArrayList<>();
+            public void createFields(
+                LuceneDocument document,
```

Comment:
I would prefer to pass in an explicit interface or alternatively `Consumer<Field>`.

Reply:
I thought the same initially https://github.com/elastic/elasticsearch/pull/85688#pullrequestreview-931539042 :) But I think I agree with Ievgen and prefer the current solution now. What's the point of adding that kind of indirection here? It may look a little nicer in isolation, but makes the big picture of what this method does less obvious (which makes reasoning about performance harder and invites problems like the intermediary list we remove here). Also, while the compiler can mostly figure this kind of thing out and make it fully inline, I'd rather not push luck on that front either and rather make the performance obvious.